src/parser/parseSVGDocument.ts

```diff
@@ -74,22 +85,33 @@ export function parseSVGDocument(
         (el) => svgValidTagNamesRegEx.test(el.nodeName.replace('svg:', ''))
       );
     });
+
+  // thos are like globals we need to fix
   gradientDefs[svgUid] = getGradientDefs(doc);
   cssRules[svgUid] = getCSSRules(doc);
   clipPaths[svgUid] = localClipPaths;
+
   // Precedence of rules:   style > class > attribute
-  parseElements(
+  const elementParser = new ElementsParser(
     elements,
-    (instances, elements) => {
-      if (callback) {
-        callback(instances, options, elements, descendants);
-        delete gradientDefs[svgUid];
-        delete cssRules[svgUid];
-        delete clipPaths[svgUid];
-      }
-    },
-    { ...options },
+    options,
```

Comment:
was there are reason it was spread before?

Reply:
ah good point i don;t know.